File: ruby/muse/algorithms/search/LinearSearchTest.rb
require "muse/algorithms/search/LinearSearch"
require "muse/util/SequenceBuilder"
require "muse/util/TestRunner"

def testLinearSearch
    size = 32768

    arr = Array.new(size)
    SequenceBuilder.packIncreasing(arr)

    if LinearSearch.find(arr, -1) != size
        return false
    end

    if LinearSearch.find(arr, 2_147_483_647) != size
        return false
    end

    arr.each_index do |i|
        if LinearSearch.find(arr, arr[i]) != i
            return false
        end
    end

    true
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.pick(testLinearSearch())
end

File: python/muse/algorithms/search/linear_search_test.py
from muse.algorithms.search import linear_search, tests
from muse.util import test_runner


def test_linear_search() -> bool:
    find = linear_search.find
    return tests.derive_empty(find)() and tests.derive_increasing(find, 10000)()


def main() -> None:
    test_runner.pick(test_linear_search)


if __name__ == "__main__":
    main()
